<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BuilderMainWindowBase</class>
 <widget class="QMainWindow" name="BuilderMainWindowBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>795</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>3</number>
    </property>
    <property name="spacing">
     <number>3</number>
    </property>
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="widget_3" native="true">
       <property name="maximumSize">
        <size>
         <width>360</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QTreeWidget" name="treeWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string notr="true">1</string>
            </property>
           </column>
          </widget>
          <widget class="QStackedWidget" name="stackedWidget">
           <property name="minimumSize">
            <size>
             <width>300</width>
             <height>300</height>
            </size>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="page0">
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0">
              <widget class="BodyPartEditor" name="partsWidget" native="true"/>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page1">
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="BodyJointEditor" name="jointWidget" native="true">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="margin">
         <number>1</number>
        </property>
        <property name="spacing">
         <number>1</number>
        </property>
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>modelview</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QSplitter" name="splitter_2">
              <property name="midLineWidth">
               <number>1</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <widget class="wombat::GLRenderWidget" name="renderWidget" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>messages</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="wombat::QMessageWriter" name="messageWidget"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionAddNode"/>
    <addaction name="actionDeleteNode"/>
    <addaction name="separator"/>
    <addaction name="actionUpdateModel"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionUpdateModel"/>
   <addaction name="separator"/>
   <addaction name="actionAddNode"/>
   <addaction name="actionDeleteNode"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/document-open.png</normaloff>:/icons/24/document-open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/document-save.png</normaloff>:/icons/24/document-save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/document-save-as.png</normaloff>:/icons/24/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>&amp;save as</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/document-new.png</normaloff>:/icons/24/document-new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;new</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/system-shutdown.png</normaloff>:/icons/24/system-shutdown.png</iconset>
   </property>
   <property name="text">
    <string>&amp;close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionUpdateModel">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/view-refresh.png</normaloff>:/icons/24/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>update Model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionAddNode">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/list-add.png</normaloff>:/icons/24/list-add.png</iconset>
   </property>
   <property name="text">
    <string>add Node</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionDeleteNode">
   <property name="icon">
    <iconset resource="../../src/wombat/resources/wombat.qrc">
     <normaloff>:/icons/24/list-remove.png</normaloff>:/icons/24/list-remove.png</iconset>
   </property>
   <property name="text">
    <string>delete Node</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>wombat::GLRenderWidget</class>
   <extends>QWidget</extends>
   <header location="global">wombat/ui.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>BodyPartEditor</class>
   <extends>QWidget</extends>
   <header location="global">widgets/BodyPartEditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>BodyJointEditor</class>
   <extends>QWidget</extends>
   <header location="global">widgets/BodyJointEditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>wombat::QMessageWriter</class>
   <extends>QListWidget</extends>
   <header location="global">wombat/helpers.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../src/wombat/resources/wombat.qrc"/>
 </resources>
 <connections/>
 <slots>
  <signal>partDataChanged()</signal>
  <signal>jointDataChanged()</signal>
 </slots>
</ui>
